Mathlib/Algebra/MvPolynomial/Basic.lean

```diff
@@ -875,6 +875,34 @@ theorem C_dvd_iff_dvd_coeff (r : R) (φ : MvPolynomial σ R) : C r ∣ φ ↔ 
     IsRegular (∏ n in s, X n : MvPolynomial σ R) :=
   IsRegular.prod fun _ _ ↦ isRegular_X
 
+/-- The finset of nonzero coefficients of a multivariate polynomial. -/
+def coeffs (p : MvPolynomial σ R) : Finset R :=
+  letI := Classical.decEq R
+  Finset.image p.coeff p.support
+
+lemma coeffs_zero : coeffs (0 : MvPolynomial σ R) = ∅ :=
+  rfl
+
+lemma coeffs_one : coeffs (1 : MvPolynomial σ R) ⊆ {1} := by
+  classical
+    simp_rw [coeffs, Finset.image_subset_iff]
+    simp_all [coeff_one]
+
+@[simp]
+lemma coeffs_one_of_nontrivial [Nontrivial R] : coeffs (1 : MvPolynomial σ R) = {1} := by
+  apply Finset.Subset.antisymm coeffs_one
+  simp only [coeffs, Finset.singleton_subset_iff, Finset.mem_image]
+  exact ⟨0, by simp⟩
+
```

Comment:
Can you add `coeffs S = ∅` if `Subsingleton R` and tag it with `nontriviality`? And also `zero_not_mem_coeffs` might be useful?

Reply:
I added the two `lemma`s.